VideoLocker/src/test/java/org/edx/mobile/view/CourseUnitNavigationActivityTest.java

```diff
@@ -413,142 +413,157 @@ public void testUnitFragmentCreation() {
+                when(videoData2.encodedVideos.getYoutubeVideoInfo())
+                        .thenReturn(Mockito.mock(VideoInfo.class));
+                when(youtubeVideosModel.getData()).thenReturn(videoData2);
+                argsList.add(new Object[] {youtubeVideosModel, CourseUnitOnlyOnYoutubeFragment.class, true});
+
+                DiscussionBlockModel discussionModel = Mockito.mock(DiscussionBlockModel.class);
+                argsList.add(new Object[] {discussionModel, CourseUnitDiscussionFragment.class, true});
+                argsList.add(new Object[] {discussionModel, CourseUnitMobileNotSupportedFragment.class, false});
+
+                CourseComponent nonMultiDeviceModel = Mockito.mock(CourseComponent.class);
+                when(nonMultiDeviceModel.isMultiDevice()).thenReturn(false);
+                argsList.add(new Object[] {nonMultiDeviceModel, CourseUnitMobileNotSupportedFragment.class, true});
+
+                HtmlBlockModel htmlModel = Mockito.mock(HtmlBlockModel.class);
+                when(htmlModel.isMultiDevice()).thenReturn(true);
+                when(htmlModel.getType()).thenReturn(BlockType.HTML);
+                argsList.add(new Object[] {htmlModel, CourseUnitWebViewFragment.class, true});
+
+                CourseComponent unknownModel = Mockito.mock(CourseComponent.class);
+                when(unknownModel.isMultiDevice()).thenReturn(true);
+                when(unknownModel.getType()).thenReturn(BlockType.COURSE);
+                argsList.add(new Object[] {unknownModel, CourseUnitEmptyFragment.class, true});
+
+                CourseComponent problemModel = Mockito.mock(CourseComponent.class);
+                when(problemModel.isMultiDevice()).thenReturn(true);
+                when(problemModel.getType()).thenReturn(BlockType.PROBLEM);
+                argsList.add(new Object[] {problemModel, CourseUnitMobileNotSupportedFragment.class, true});
+
+                CourseComponent othersModel = Mockito.mock(CourseComponent.class);
+                when(othersModel.isMultiDevice()).thenReturn(true);
+                when(othersModel.getType()).thenReturn(BlockType.OTHERS);
+                argsList.add(new Object[] {othersModel, CourseUnitMobileNotSupportedFragment.class, true});
+            }
+
+            return argsList;
         }
 
+        /**
+         * The {@link CourseComponent} that we provide to the {@link CourseUnitPagerAdapter} as input.
+         */
         @Parameterized.Parameter
-        public CourseComponent input;
+        public CourseComponent inputCourseComponent;
 
+        /**
+         * The class of the {@link Fragment} that we expect the {@link CourseUnitPagerAdapter} will
+         * create an instance of.
+         */
         @Parameterized.Parameter(value = 1)
-        public Class expected;
+        public Class expectedFragmentClass;
 
+        /**
+         * <code>true</code> if we want to enable discussions for the test, <code>false</code>
+         * otherwise.
```

Comment:
This would be better as a direct description of the variable (instead of just describing the expected value), e.g. "A flag parameter denoting whether inline discussions should be enabled in the config.".